recipes/openjdk/all/test_package/conanfile.py

```diff
@@ -0,0 +1,18 @@
+from conans import ConanFile
+from six import StringIO
+
+
+class TestPackage(ConanFile):
+
+    def build(self):
+        pass # nothing to build, but tests should not warn
+
+    def test(self):
+        test_cmd = ['java', '--version']
+        output = StringIO()
+        self.run(test_cmd, output=output)
+        version_info = output.getvalue()
+        if "openjdk" in version_info:
+            pass
+        else:
+            raise Exception("java call seems not use the openjdk bin")
```

Comment:
https://docs.conan.io/en/latest/devtools/build_requires.html#testing-build-requires
I think it's ok to import from `io`.

```suggestion
from conans import ConanFile
from conans.errors import ConanException
from io import StringIO

required_conan_version = ">=1.36.0"


class TestPackage(ConanFile):
    test_type = "build_requires"

    def build(self):
        pass # nothing to build, but tests should not warn

    def test(self):
        output = StringIO()
        self.run("java --version", output=output, run_environment=True)
        print(output.getvalue)
        version_info = output.getvalue()
        if "openjdk" in version_info:
            pass
        else:
            raise ConanException("java call seems not use the openjdk bin")
```